### ISSUE TITLE
`no-useless-else`: False positive after conditional `throw`, autofix breaks code

### ISSUE THREAD

**Issue by ehoogeveen-medweb** [sindresorhus/eslint-plugin-unicorn#3607]
After updating to version `73.0.0`, I'm getting a false positive in `no-useless-else`, where it appears to identify a conditional `throw` statement as an unconditional exit.

The autofix makes the code in the `else` branch unconditional, breaking the logic.

Buggy rule: `no-useless-else`
First broken version: `73.0.0` (no false positive in `72.0.0`)

Reduced example:
```js
const MODE_GROUP = new Set(['foo', 'bar']);

const args = new Map();

const modes = new Set();
if (args.has('modes')) {
  for (const mode of args.get('modes').split(',')) {
    const trimmed = mode.trim();
    if (MODE_GROUP.has(trimmed)) {
      modes.add(trimmed);
    }
  }
  if (!modes.size) {
    throw new Error(`No valid modes in 'modes'!`);
  }
  // eslint-disable-next-line unicorn/no-useless-else -- False positive; we only throw if there are no valid modes.
} else {
  for (const mode of MODE_GROUP) {
    modes.add(mode);
  }
}
```

**Comment by ehoogeveen-medweb** [comment]
Here's a more minimal testcase:
```javascript
const set = new Set();

if (globalThis.condition) {
  set.add('a');
  if (!set.size) {
    throw new Error('empty');
  }
} else {
  globalThis.fallback();
}
```

AI gives the following root cause analysis:

In `isBranchExit` ([`rules/utils/is-branch-exit.js:1816-1821`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/2e19a9401d84cc0d2f7f73acc33b4e4f154685de/rules/utils/is-branch-exit.js#L1816)), an `IfStatement` whose test has a computable static value is collapsed to the selected branch. `getStaticValue` from `@eslint-community/eslint-utils` evaluates `set.size` on a `const set = new Set()` as `0`, so `!set.size` folds to `true`. The rule then treats the inner `if` as an unconditional `throw` and reports the outer `else`.

The bug is that the constant folding ignores intervening mutation: `set.add('a')` runs before the test, so `set.size` is not `0` at that point. The `Set` (and its `.size`) is what triggers it — the same shape with a plain object (`const object = {n: 0}; object.n = 1; if (!object.n)`) does **not** reproduce, because `getStaticValue` won't fold object property reads.

**Comment by ehoogeveen-medweb** [comment]
From further analysis, it seems to be a bug in @eslint-community/eslint-utils where `getStaticValue` doesn't properly handle `Set` or `Map`.

I filed https://github.com/eslint-community/eslint-utils/issues/353
